components/combobox/combobox.test.js

```diff
@@ -175,55 +162,48 @@ describe('DtCombobox Tests', () => {
     });
 
     describe('When it is loading', () => {
-      // Test Setup
-      beforeEach(async () => {
-        slots = { list: '<ol id="list"><li role="option">item1</li><li role="option">item2</li></ol>' };
-        await wrapper.setProps({ loading: true });
-        _setChildWrappers();
-      });
+      it('should render the loading skeletons', () => {
+        mockSlots = { list: '<ol id="list"><li role="option">item1</li><li role="option">item2</li></ol>' };
+        mockProps = { loading: true };
+
+        updateWrapper();
 
-      it(
-        'should render the loading skeletons',
-        () => { expect(skeletons.exists()).toBe(true); },
-      );
+        expect(skeletons.exists()).toBe(true);
+      });
     });
   });
 
   describe('Accessibility Tests', () => {
     describe('When a input is provided', () => {
-      // Test Setup
-      beforeEach(async () => {
-        scopedSlots = { input: '<input v-bind="props.inputProps" />' };
-        _mountWrapper();
-        _setChildWrappers();
+      beforeEach(() => {
+        mockScopedSlots = { input: '<input v-bind="props.inputProps" />' };
+
+        updateWrapper();
       });
 
       describe('When list is not expanded', () => {
-        beforeEach(async () => {
+        it('aria-expanded should be "false"', async () => {
           await wrapper.setProps({ showList: false });
```

Comment:
This should be changed to:

```suggestion
          mockProps = { showList: false };
          
          updateWrapper();
```

and remove the async

Reply:
This way of do it is valid and is specified on [TEST_GUIDELINE](https://github.com/dialpad/dialtone-vue/blob/staging/.github/TEST_CONTRIBUTING/TEST_GUIDELINE.md).

`You can utilize mockProps to set environment configuration to the mounted component, and use .setProps method if you want to trigger some component behaviour.`

The difference on doing it using `updateWrapper();` or `.setProps()` is in the first one you will re-mount the component, on the second one it will react to a change.

Your suggestion on this is valid.

As you will see on the change, I removed the `beforeEach` (because it is only one test) and just reuse the `setProps` because there is no need to update it.

Reply:
You're right, I didn't see that on the guideline, I'd be good to make that more visible.